### PR TITLE
airoah: real watchdog + trng

### PR DESCRIPTION
**Pull request by Ansuel**

### CODE REVIEW

File: target/linux/airoha/dts/en7581.dtsi [hunk 1]
```diff
@@ -215,6 +215,12 @@
 			#reset-cells = <1>;
 		};
 
+		rng@1faa1000 {
+			compatible = "airoha,en7581-trng";
+			reg = <0x1faa1000 0x1000>;
+			interrupts = <GIC_SPI 19 IRQ_TYPE_LEVEL_HIGH>;
```
Comment (merbanan) [inline target/linux/airoha/dts/en7581.dtsi]: This is most likely wrong. Add 16.

**Comment by Ansuel** [comment]
@LorenzoBianconi trng + watchdog tested and works correctly (trng with the rng tools utility, watchdog by modify the driver to disable ping and system correctly reboots)

airoha parser is also implemented with the art partition autocalculated.

I verified the nodes match the original firmware of the reference board. I also attach and enable the bad block driver as it does reduce the total NAND partition. Also about that we were missing some bits about that.

I think the NAND require still some stuff as the FIT image are still not correctly detected (maybe we are missing some specific tweaks about them?)

**Comment by merbanan** [comment]
Not related to the NAND partitions but BMT. If the BMT structure on NAND is not detected the ATF part in the boot loader will initialize the structure on flash and mark its blocks as bad. In the SDK the drivers can remap blocks via the BMT logic. It has some slight use in the case in raw partitions on NAND. But anything with its own wear-leveling support will lead to improper wear statistics in the case of a block remap.

It is possible to patch away the BMT logic in the ATF but if a remap has been done in the boot loader area it will not be able to boot.

To cleanly get rid of BMT in all scenarios seem impossible.

**Comment by Ansuel** [comment]
@merbanan problem is that on top of this, the current logic done by the parser (with the art partition placed at the end of the flash) account for bmt soo excluding bmt support results in the wrong offset calculated.

Current implementation reflect the same offset of the sdk

**Pull request merged**